libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -148,12 +148,13 @@ method rpcHandler*(f: FloodSub,
       discard
 
     var toSendPeers = initHashSet[PubSubPeer]()
-    for t in msg.topicIds:                     # for every topic in the message
-      if t notin f.topics:
-        continue
-      f.floodsub.withValue(t, peers): toSendPeers.incl(peers[])
+    let topic = msg.topicId
+    if topic notin f.topics:
+      continue
+    f.floodsub.withValue(topic, peers):
+      toSendPeers.incl(peers[])
```

Comment:
Hum... it looks wrong to me. The `continue` was for the loop you removed. Something like that should fix it.
```suggestion
    if topic in f.topics:
      f.floodsub.withValue(topic, peers):
        toSendPeers.incl(peers[])
      await handleData(f, topic, msg.data)
```

Reply:
Nevermind, `f.broadcast` will do nothing if `topic` isn't in `f.topics`

Reply:
Could you please share why `f.broadcast` won't do anything? Even if it's true, IMO we shouldn't rely on that and change the previous behavior without a good reason for doing so.

Reply:
Because `broadcast` [will update some metrics](https://github.com/vacp2p/nim-libp2p/blob/unstable/libp2p/protocols/pubsub/pubsub.nim#L170-L210) by incrementing by the number of `toSendPeers` which is zero.
Then it iterates through `toSendPeers` [to send data](https://github.com/vacp2p/nim-libp2p/blob/unstable/libp2p/protocols/pubsub/pubsub.nim#L212-L219) (encoded or not) but it is empty so it won't do anything.

Reply:
Thanks, you are right. But it'd still log. Maybe adding a log when `topic` isn't in `f.topics` is enough.

Reply:
Added